src/sources/datadog_agent/logs.rs

```diff
@@ -111,55 +112,55 @@ pub(crate) fn decode_log_body(
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("status"),
+                                Some(LegacyKey::InsertIfEmpty(path!("status"))),
                                 path!("status"),
                                 status.clone(),
                             );
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("timestamp"),
+                                Some(LegacyKey::InsertIfEmpty(path!("timestamp"))),
                                 path!("timestamp"),
                                 timestamp,
                             );
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("hostname"),
+                                Some(LegacyKey::InsertIfEmpty(path!("hostname"))),
                                 path!("hostname"),
                                 hostname.clone(),
                             );
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("service"),
+                                Some(LegacyKey::InsertIfEmpty(path!("service"))),
                                 path!("service"),
                                 service.clone(),
                             );
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("ddsource"),
+                                Some(LegacyKey::InsertIfEmpty(path!("ddsource"))),
                                 path!("ddsource"),
                                 ddsource.clone(),
                             );
                             namespace.insert_source_metadata(
                                 source_name,
                                 log,
-                                path!("ddtags"),
+                                Some(LegacyKey::InsertIfEmpty(path!("ddtags"))),
                                 path!("ddtags"),
                                 ddtags.clone(),
                             );
 
                             namespace.insert_vector_metadata(
                                 log,
-                                path!(source.log_schema_source_type_key),
```

Comment:
Is it just not necessary to macro this, the type already matches?

Reply:
This was previously a mistake. The log schema keys are supposed to be full keys (may contain multiple segments). The macro was forcing it to be a single segment. (It's still being passed in as a string now, but that will improve in the future)